### DISCUSSION TITLE
Aqara QBKG12LM switch delay

### DISCUSSION THREAD

**Discussion by OleksandrMasliuchenkoGL** [Koenkk/zigbee2mqtt#5922] [Devices]
Hi All,

I have 2 very similar combinations of aqara switches, which unfortunately behave very differently.

Case 1: 2 button wireless wall switch WXKG02LM controls lights connected to another wired switch QBKG12LM. Logical connection is done using Home Assistant automation. This set up works perfectly, when I click a button on the wireless switch, wired switch toggles almost immediately (with no human noticeable delay)

Case 2: 2 button wired switch QBKG12LM, one of the button is decoupled from its relay and just sends click messages. It is logically (with another HA automation) controls one more QBKG12LM wired switch.

Unfortunately there is noticeable delay about 1 second between clicking the button, and lights turn on. I tried to look at z2m logs, and it looks like the actual delay happens between clicking the button, and getting a corresponding log record - everything else happens immediately.

Does someone know how to improve QBKG12LM delay?

**Comment by llrp** [comment]
Did you manage to resolve this? Seems to be an ongoing issue

**Reply by OleksandrMasliuchenkoGL** [reply]
Nope, I started writing my own firmware instead

**Reply by llrp** [reply]
> OleksandrMasliuchenkoGL

Any success with aforementioned self-authored firmware?
